vertexai/token-count/tokencount.go

```diff
@@ -42,13 +42,19 @@ func countTokens(w io.Writer, prompt, projectID, location, modelName string) err
 
 	model := client.GenerativeModel(modelName)
 
-	resp, err := model.CountTokens(ctx, genai.Text(prompt))
+	resp, err := model.CountTokens(ctx, prompt)
 	if err != nil {
 		return err
 	}
 
 	fmt.Fprintf(w, "Number of tokens for the prompt: %d\n", resp.TotalTokens)
 
+	resp2, err := model.GenerateContent(ctx, prompt)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Total number of tokens for the prompt: %d\n", resp2.UsageMetadata.PromptTokenCount)
```

Comment:
question: Does `resp.TotalTokens = resp2.UsageMetadata.PromptTokenCount`? 
nit: I think we should add a comment or update the printed statement to reflect the difference between these two numbers. My understanding is that `resp.TotalTokens` is calculated without any generation, while `resp2.UsageMetadata.PromptTokenCount` is post generation and is the actual billed amount of tokens. Correct?

Reply:
Yes, the number is the same, cause `CountTokens()` accepts prompt and calculate input number of tokens.
You are right `resp2.UsageMetadata` contains the data about how many tokens were passed - `PromptTokenCount` and number of tokens being generated - `CandidatesTokenCount`. The billed amount for the input is the same, cause in both calls accepts the same value, however output text is also billed, but this number can't be estimated.

I guess this is just a proto attributes naming confusion, cause `CountTokens()` total means tokens just for input, but for `GenerateContent()` total means input + output (`PromptTokenCount + CandidatesTokenCount`)

I will add print for the output part as well, to make it more clear.